src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -184,6 +184,53 @@ class RAY_EXPORT GcsClient : public std::enable_shared_from_this<GcsClient> {
   std::function<void()> resubscribe_func_;
 };
 
+// This client is only supposed to be used from Cython / Python
+class RAY_EXPORT GcsSyncClient {
+ public:
+  explicit GcsSyncClient(const GcsClientOptions &options);
+  Status Connect();
+
+  Status InternalKVGet(const std::string &ns,
```

Comment:
So I actually considered this -- the current interface of `InternalKVAccessor` seemed more appropriate for an asynchronous interface and the methods that were needed here were really simple to implement, so I discarded the idea. For the methods that do exist in the InternalKVAccessor and are synchronous, the basically all have the wrong signature (missing timeouts, missing information about the actions that have been taken for the updates or having the wrong types like bool vs. int). If you strongly feel the `InternalKVAccessor` should be reused, I could probably make that happen, but I tried to not modify Ray internals as part of this PR to keep it more incremental and risk-free.

I also first tried to reuse the whole existing C++ GCS client but that is basically impossible since it is deeply coupled with the async implementation.

Reply:
> If you strongly feel the InternalKVAccessor should be reused, I could probably make that happen, but I tried to not modify Ray internals as part of this PR to keep it more incremental and risk-free.

Gotcha - no, i don't have a strong preference for reusing that one. I guess my intuition was these two could be unified. 

So it sounds like maybe in a future PR, we could actually refactor the internal KV accessor to have the proper signatures, and we could unify the imlementation?

Reply:
Yeah, that sounds great -- whether this is worth doing also depend on whether this client is going to be used from C++ in the future. If it is, the value of doing the unification is definitely higher :)